azurelinuxagent/common/utils/archive.py

```diff
@@ -240,6 +231,43 @@ def save(self, data, file_name):
                 self._errors = True
                 logger.warn("Failed to save {0} to the goal state history: {1} [no additional errors saving the goal state will be reported]".format(file_name, e))
 
+    _purge_error_count = 0
+
+    @staticmethod
+    def _purge():
+        """
+        Delete "old" history directories and .zip archives. Old is defined as any directories or files older than the X newest ones.
+        """
+        try:
+            history_root = os.path.join(conf.get_lib_dir(), ARCHIVE_DIRECTORY_NAME)
+
+            if not os.path.exists(history_root):
+                return
+
+            items = []
+            for current_item in os.listdir(history_root):
+                full_path = os.path.join(history_root, current_item)
+                items.append(full_path)
+            items.sort(key=os.path.getctime, reverse=True)
+
+            for current_item in items[_MAX_ARCHIVED_STATES:]:
+                if os.path.isfile(current_item):
+                    os.remove(current_item)
+                else:
+                    shutil.rmtree(current_item)
+
+            if GoalStateHistory._purge_error_count > 0:
```

Comment:
I think it should be `>=0`. Happy path scenario it will be 0 and still want to log `successfully cleaned up `message

Reply:
The intention is to log this message only when we are recovering from a previous error (I added a comment to clarify this).

We don't want to log all successful calls, since we could be logging every 6 seconds if we hit an scenario similar to the one reported in #2619